src/index.ts

```diff
@@ -159,58 +168,23 @@ export class Supertab {
     } catch (e) {}
     /* eslint-enable */
 
-    const presentedCurrency =
-      tab?.currency.isoCode ??
-      preferredCurrencyCode ??
-      clientConfig.suggestedCurrency ??
-      DEFAULT_CURRENCY;
-
-    const currenciesByCode: Record<string, Currency> =
-      clientConfig.currencies.reduce(
-        (acc, eachCurrency) => ({
-          ...acc,
-          [eachCurrency.isoCode]: eachCurrency,
-        }),
-        {},
-      );
-
-    const getPrice = (
-      offering: SiteOffering,
-      price: Price,
-      currencyCode?: string,
-    ) => {
-      const currency =
-        currenciesByCode[currencyCode ?? price.currency] ??
-        currenciesByCode[DEFAULT_CURRENCY];
-
-      const text = formatPrice({
-        amount: offering.price.amount,
-        currency: currency.isoCode,
-        baseUnit: currency.baseUnit,
-        localeCode: language,
-        showZeroFractionDigits: true,
-        showSymbol: currency.isoCode !== "CHF",
+    const { presentedCurrency, currenciesByCode, getPrice } =
+      await setupCurrencyHandling({
+        tab,
+        preferredCurrencyCode,
+        currencies: clientConfig.currencies,
+        suggestedCurrency: clientConfig.suggestedCurrency,
+        language,
       });
 
-      return {
-        amount: offering.price.amount,
-        currency: getPublicCurrencyDetails(currency),
-        text,
-      };
-    };
-
     const offerings = clientConfig.offerings
       .filter((eachOffering) => !!currenciesByCode[eachOffering.price.currency])
-      // Format all offerings
       .map((eachOffering) => {
         const prices = eachOffering.prices?.map((eachPrice) =>
-          getPrice(eachOffering, eachPrice),
+          getPrice(eachPrice),
         );
 
         let connectedSubscriptionOffering;
-        // If the offering has a connected subscription offering, store the id
-        // temporarily to be able to find the formatted connected subscription
-        // offering in the next map function.
```

Comment:
I'm wondering if this and similar comments are worth keeping to help our future selves understand what this code does. Is there a particular reason why this PR removes those comments?

Reply:
Mistake on my part. Thank you for flagging!